### DISCUSSION TITLE
Week 5 - Refactoring & New Map

### DISCUSSION THREAD

**Discussion by Mohmad-Naser-alnakeshbandi** [The-Escape-Box/TheEscapeBox#31] [General]
This week, our attention is firmly set on two crucial aspects shaping the evolution of our game.

## Refactoring:
We're diving deep into our existing codebase, meticulously evaluating its functionalities while actively seeking areas for improvement. Through this process of refinement, our goal is to boost performance and establish a robust framework for forthcoming development phases.

##  New Map:
Exciting news! We've just wrapped up the creation of a brand-new map set within a hospital environment. This addition promises to deliver fresh, immersive experiences for our players, opening up exciting new avenues for exploration and gameplay.
![image](https://github.com/The-Escape-Box/TheEscapeBox/assets/76824323/bfa9d77b-4d9b-47b7-bec4-b2457c5913ab)

## Machine Gun Addition:
We've taken your feedback to heart, and we're thrilled to announce the introduction of a new weapon – the Machine Gun – to our arsenal. Complete with a dynamic particle system and sleek shooting animations, this addition is sure to elevate the intensity and excitement of gameplay.
![image](https://github.com/The-Escape-Box/TheEscapeBox/assets/76824323/0a5a0f2b-b497-4132-97e6-78a5fc13ea45)

## Time spend
Time we spent this week:
Mohamad: 25h
Lukas: 5h

Feel free to try out the game and tell us about bugs or features you would love to see in the game. We will read all your opinions and try to make decision all of you love.

**Comment by Claiyc** [comment]
Great blog post! It's fantastic to see your team's dedication to refining the game through code refactoring and the addition of a new map set in a hospital environment. The introduction of the Machine Gun is also exciting news! How do you envision players getting hold of this new weapon, will they be able to find it on the map or will they have to buy it through your ingame interface?

**Reply by WeberWr** [reply]
Thanks for your feedback. We also thought about that question and decided to make the Machine Gun the new main weapon of the player. So the pistol is not longer in the game. We made that decision because we think this will make it more easy to balance the game and we don´t need to animate the pistol correct.
